src/Nethermind/Nethermind.Synchronization/StateSync/StateSyncDispatcher.cs

```diff
@@ -32,11 +38,42 @@ public StateSyncDispatcher(ISyncFeed<StateSyncBatch> syncFeed, ISyncPeerPool syn
         {
         }
 
-        protected override async Task Dispatch(PeerInfo peerInfo, StateSyncBatch request, CancellationToken cancellationToken)
+        protected override async Task Dispatch(PeerInfo peerInfo, StateSyncBatch batch, CancellationToken cancellationToken)
         {
+            if (batch?.RequestedNodes == null || batch.RequestedNodes.Length == 0)
+            {
+                return;
+            }
+
             ISyncPeer peer = peerInfo.SyncPeer;
-            var getNodeDataTask = peer.GetNodeData(request.RequestedNodes.Select(n => n.Hash).ToArray(), cancellationToken);
-            await getNodeDataTask.ContinueWith(
+            Keccak[]? a = batch.RequestedNodes.Select(n => n.Hash).ToArray();
+            Task<byte[][]> task = null;
+
+            // Use GETNODEDATA if possible
+            // TODO: Currently all SyncPeer support ETH66 but it's going to change
+            // so we need to introduce some IF here to check ETH66 support
+            task = peer.GetNodeData(a, cancellationToken);
+
+            // GETNODEDATA is not supported so we try with SNAP protocol
+            if (task is null && peer.TryGetSatelliteProtocol<ISnapSyncPeer>("snap", out ISnapSyncPeer handler))
```

Comment:
`task is null`?

Is that implemented/intended or did you missed an `await`?

Reply:
There is TODO where @marcindsobczak is going to add IF clause in #4506.